mantidproject/mantid
# Implement fourier filter to Polaris.create_total_scattering_pdf

This PR adds a way to perform a fourier filter as part of the create_total_scattering_pdf, this can remove signals bellow and above certain boundaries given as a list paramiter freq_params.
It also fixes a bug that prevented lists being used for q_limits.

**Report to:** Helen Playford

**To test:**
I can provide the config files required for the script below.
test running the script below with different paramiters for `create_total_scattering_pdf`, the `pdf_output` workspace should have no peaks outside of the range specified by `freq_params` and the Qspace workspace should also be smoothed and fluctute around zero at larger Q
```
from mantid.simpleapi import *
from isis_powder import polaris, SampleDetails

config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="Test", mode="PDF")  
sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si')
polaris.set_sample_details(sample=sample_details)
polaris.create_vanadium(first_cycle_run_no="98532",
                        multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = [[2.5, 3, 4, 6, 7], [3.5, 5, 7, 11, 40]]
polaris.create_total_scattering_pdf(run_number="98533", 
                                    merge_banks=True, 
                                    q_lims=q_lim, 
                                    output_binning=[0,0.1,20],
                                    pdf_type='G(r)', 
                                    freq_params=[1])
```

Fixes #27773 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
